Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/EMStudioSDK/Source/RenderPlugin/RenderPlugin.cpp

```diff
@@ -68,13 +68,11 @@ namespace EMStudio
         SaveRenderOptions();
         CleanEMStudioActors();
 
-        // Get rid of the OpenGL view widgets.
-        // Don't delete them directly as there might be still paint events in the Qt message queue which will cause a crash.
-        // deleteLater will make sure all events will be processed before actually destructing the object.
         for (RenderViewWidget* viewWidget : m_viewWidgets)
         {
-            viewWidget->deleteLater();
+            delete viewWidget;
```

Comment:
This is cool to remove `deleteLater()` but what is protecting paint events now from accessing these widgets after deletion?

Reply:
The comment that was there just incorrectly describes the previous situation. The commit message sums it up properly:

The previous code was attempting to call `deleteLater()` to delete a child
widget, in case there were pending events on the event queue that would be
processed later and attempt to call a method on an deleted object. However,
as part of the base `QWidget` destructor, it destroys all child widgets, so
the child RenderViewWidget is deleted as part of the plugin's destructor
anyway. The issue is that the RenderViewWidget maintains a pointer to the
plugin, and attempts to unregister itself from the plugin in its
destructor. But by the time that the RenderViewWidget's destructor runs,
the plugin's destructor has already completed, so the call to unregister
itself is using an already deleted pointer.

The solution is to delete the RenderViewWidgets directly in the plugin's
destructor, so that when the RenderViewWidget unregisters itself from the
plugin, the plugin pointer is still valid.